backend/src/Designer/EventHandlers/ProcessTaskIdChanged/ProcessTaskIdChangedLayoutSetsHandler.cs

```diff
@@ -23,44 +27,128 @@ public ProcessTaskIdChangedLayoutSetsHandler(IAltinnGitRepositoryFactory altinnG
-            notification.EditingContext,
-            SyncErrorCodes.LayoutSetsTaskIdSyncError,
-            "App/ui/layout-sets.json",
-            async () =>
-            {
-                var repository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(
-                    notification.EditingContext.Org,
-                    notification.EditingContext.Repo,
-                    notification.EditingContext.Developer);
+        var repository = _altinnGitRepositoryFactory.GetAltinnAppGitRepository(
+            notification.EditingContext.Org,
+            notification.EditingContext.Repo,
+            notification.EditingContext.Developer);
+
+        if (!repository.AppUsesLayoutSets())
+        {
+            return;
+        }
+
+        await ProcessLayoutSets(notification, repository, cancellationToken);
+    }
 
-                if (!repository.AppUsesLayoutSets())
+    private async Task ProcessLayoutSets(ProcessTaskIdChangedEvent notification,
+        AltinnAppGitRepository repository, CancellationToken cancellationToken)
+    {
+        var layoutSetsFile = await repository.GetLayoutSetsFile(cancellationToken);
+
+        foreach (string layoutSetName in layoutSetsFile.Sets.Select(layoutSet => layoutSet.Id))
+        {
+            await _fileSyncHandlerExecutor.ExecuteWithExceptionHandlingAndConditionalNotification(
+                notification.EditingContext,
+                SyncErrorCodes.LayoutSetsTaskIdSyncError,
+                "App/ui/layout-sets.json",
+                async () =>
                 {
-                    return hasChanges;
-                }
+                    bool hasChanged = false;
+
+                    var layoutSets = await repository.GetLayoutSetsFile(cancellationToken);
+                    if (TryChangeLayoutSetTaskIds(layoutSets, notification.OldId, notification.NewId))
+                    {
+                        await repository.SaveLayoutSets(layoutSets);
+                        hasChanged = true;
+                    }
 
-                var layoutSets = await repository.GetLayoutSetsFile(cancellationToken);
-                if (TryChangeTaskIds(layoutSets, notification.OldId, notification.NewId))
+                    return hasChanged;
+                });
+
+            await ProcessLayouts(layoutSetName, notification, repository, cancellationToken);
+        }
+    }
+
+    private async Task ProcessLayouts(string layoutSetName, ProcessTaskIdChangedEvent notification,
+        AltinnAppGitRepository repository, CancellationToken cancellationToken)
+    {
+        string[] layoutNames;
+        try
+        {
+            layoutNames = repository.GetLayoutNames(layoutSetName);
+        }
+        catch (FileNotFoundException)
+        {
+            return;
+        }
+
+        foreach (string layoutName in layoutNames)
+        {
+            string layoutPath = $"App/ui/{layoutSetName}/{layoutName}.json";
```

Comment:
I suggest moving the new code that handles the layouts to its own file (e.g. `ProcessTaskIdChangedLayoutsHandler.cs`) to keep it separate from the layout sets logic